jupyterlab_caip_optimizer/src/store/snackbar.ts

```diff
@@ -0,0 +1,53 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { AppThunk } from './store';
+
+export interface OpenAction {
+  severity: 'error' | 'warning' | 'info' | 'success';
+  message: string;
+}
+
+export interface SnackbarState {
+  open: boolean;
+  severity: 'error' | 'warning' | 'info' | 'success';
+  message: string;
+}
+
+export const snackbarSlice = createSlice({
+  name: 'snackbar',
+  initialState: {
+    open: false,
+    message: '',
+    severity: 'info',
+  } as SnackbarState,
+  reducers: {
+    open: (state, action: PayloadAction<OpenAction>) => {
+      state.message = action.payload.message;
+      state.severity = action.payload.severity;
+      state.open = true;
+    },
+    close: state => {
+      state.open = false;
+    },
+  },
+});
+
+const wait = async (time: number) =>
+  new Promise(resolve => setTimeout(() => resolve(), time));
+
+export const createSnack = (
+  message: string,
+  severity: OpenAction['severity'] = 'info',
+  waitMs = 2000
```

Comment:
could we make `waitMs` an optional argument with default value so that we can change if we want to? I'd assume for longer error messages, 2 seconds might be a bit short. We can still have 2000ms as default.

Reply:
Correct me if I am wrong, but `waitMs = 2000` should default to 2000ms.

Reply:
https://material-ui.com/api/snackbar/ `onClose` property.

Reply:
oops sorry you're right. I didn't catch that it was inside the argument bracket.